python/pyspark/sql/tests.py

```diff
@@ -2245,11 +2245,6 @@ def test_replace(self):
                .replace(False, True).first())
         self.assertTupleEqual(row, (True, True))
 
-        # replace list while value is not given (default to None)
```

Comment:
Seems we should disallow the case. Please see https://github.com/apache/spark/pull/16793#issuecomment-362684399